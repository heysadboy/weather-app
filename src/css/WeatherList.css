#weather-list {
    padding: 30px 0px 0px 0px;
}

@media screen and (max-width: 768px) {
    
    #weather-list {
        margin: 0px!important;
        padding: 30px 0px 0px 0px;
        min-width: 100vw;
    }

    #weather-row {
        margin: 0px;
        min-width: 100vw;
    }
}