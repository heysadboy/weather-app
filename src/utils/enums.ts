export enum EActionType {
    fetch_weather = 'FETCH_WEATHER',
    temp_data = 'SET_TEMP_DATA',
    left_page = 'LEFT_PAGE',
    right_page = 'RIGHT_PAGE'
}

export enum EStatusType {
    error = 'ERROR',
    loading = 'LOADING',
    ok = 'OK'
}

export enum ETempType {
    f = "Fahrenheit",
    c = "Celsius"
}