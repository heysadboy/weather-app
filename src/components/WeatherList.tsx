import { FC } from "react";
import { connect } from "react-redux";
import { bindActionCreators } from "redux";
import { ThunkDispatch } from "redux-thunk";
import { tempData } from "../actions";
import { IWeather, IAction, IPage } from "../utils/interfaces";
import { AppState } from "../utils/types";
import '../css/WeatherList.css';
import WeatherCard from "./WeatherCard";
import { ETempType } from "../utils/enums";

interface IWeatherListProp {
    temperature: IWeather[],
    page: IPage,
    tempData: () => void,
    tempType: ETempType
}

const mapStateToProps = (state: AppState) => ({
    temperature: state.temperature,
    page: state.page
});

const mapDispatchToProps = (
    dispatch: ThunkDispatch<AppState, {}, IAction>
) => ({
    tempData: bindActionCreators(tempData, dispatch)
});

const WeatherList: FC<IWeatherListProp> = ({ page, temperature, tempType }) => {
    const renderedTempList = temperature.slice(page.start, page.end + 1).map((temp_item) => {
        return (<WeatherCard key={temp_item.id} tempType={tempType} temp_item={temp_item} />);
    });

    return (
        <div id="weather-list" className="ui three column grid container">
            <div id="weather-row" className="row">
                {renderedTempList}
            </div>
        </div>
    )
};

export default connect(mapStateToProps, mapDispatchToProps)(WeatherList);