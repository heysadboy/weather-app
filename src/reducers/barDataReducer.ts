import { EActionType } from '../utils/enums';
import { IAction } from '../utils/interfaces';


const barDataReducer = (state = [], action: IAction) => {
    switch (action.type) {
        case EActionType.barchart_data:
            return action.payload;
        default:
            return state;
    }
};

export default barDataReducer;