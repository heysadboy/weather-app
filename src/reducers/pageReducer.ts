import { EActionType } from '../utils/enums';
import { IAction } from '../utils/interfaces';
import { INITIAL_PAGE } from '../utils/data';

const pageReducer = (state = INITIAL_PAGE, action: IAction) => {
    switch (action.type) {
        case EActionType.left_page:
            if (state.start > 0) {
                return { ...state, start: state.start - 1, end: state.end - 1 };
            }
            else {
                return state;
            }

        case EActionType.right_page:
            if (state.end < 5) {
                return { ...state, start: state.start + 1, end: state.end + 1 };
            }
            else {
                return state;
            }
        default:
            return state;
    }
}

export default pageReducer;