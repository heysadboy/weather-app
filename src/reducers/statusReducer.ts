import { INITAIL_STATUS } from '../utils/data';
import { EStatusType } from '../utils/enums';
import { IAction } from '../utils/interfaces';

const statusReducer = (state = {INITAIL_STATUS}, action: IAction) => {
    switch (action.type) {
        case EStatusType.error:
            return action.payload;
        case EStatusType.ok:
            return action.payload;
        case EStatusType.loading:
            return action.payload;
        default:
            return state;
    }
}

export default statusReducer;