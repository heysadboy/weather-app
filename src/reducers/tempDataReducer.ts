import { EActionType } from '../utils/enums';
import { IAction } from '../utils/interfaces';


const tempDataReducer = (state = [], action: IAction) => {
    switch (action.type) {
        case EActionType.temp_data:
            return action.payload;
        default:
            return state;
    }
};

export default tempDataReducer;