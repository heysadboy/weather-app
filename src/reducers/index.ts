import { combineReducers } from 'redux';
import pageReducer from './pageReducer';
import tempDataReducer from './tempDataReducer';
import weatherReducer from './weatherReducer';

export default combineReducers({
    weather: weatherReducer,
    temparture: tempDataReducer,
    page: pageReducer
});