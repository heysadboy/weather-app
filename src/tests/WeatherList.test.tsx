import renderer from 'react-test-renderer';
import WeatherList from '../components/WeatherList';
import { ETempType } from '../utils/enums';
import reducers from '../reducers';
import { Provider } from 'react-redux';
import { createStore } from 'redux';;


test('Test Weather List', () => {
    const tempType: ETempType = ETempType.c;
    const currentDay: string = "test";
    const setCurrentDay = () => { };
    const store = createStore(reducers);

    const component = renderer.create(
        <Provider store={store}>
            <WeatherList tempType={tempType} setCurrentDay={setCurrentDay} currentDay={currentDay} />
        </Provider>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});