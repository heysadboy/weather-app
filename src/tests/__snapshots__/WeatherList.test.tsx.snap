// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Weather List 1`] = `
<div
  className="ui three column grid container"
  id="weather-list"
>
  <div
    className="row"
    id="weather-row"
  />
</div>
`;
